diff --git a/async/index.native.js b/async/index.native.js
index a765de9..72a287a 100644
--- a/async/index.native.js
+++ b/async/index.native.js
@@ -36,7 +36,10 @@ let customAlphabet = (alphabet, defaultSize = 21) => {
       return tick(id, size)
     })
 
-  return size => tick('', size)
+  return (size = defaultSize) => {
+    if (size <= 0) return Promise.resolve('')
+    return tick('', size)
+  }
 }
 
 let nanoid = (size = 21) =>
